database/factories/ModelFactory.php

```diff
@@ -176,6 +176,59 @@
     ]);
 });
 
+/**
+ * Empty bundle.
+ */
+$factory->define(App\Bundle::class, function (Faker\Generator $faker, $attributes) {
+
+    // get/make  registration for a family
+    $family = isset($attributes['family_id'])
+        ? App\Registration::find($attributes['family_id'])
+        : factory(App\Registration::class)->create()->family;
+
+    // get/calculate and stash the entitlement
+    $entitlement = isset($attributes['entitlement'])
+        ? $attributes['entitlement']
+        : $family->entitlement
+    ;
+
+    // Allocations: The factory won't fix an allocation without a centre
+    // If there's an allocated_at, set it.
+    $allocated_at = isset($attributes['allocated_at'])
+        ? $attributes['allocated_at']
+        : null
+    ;
+
+    // $centre is the centre we created the bundle at
+    $centre = null;
+
+    // if there is one supplied, find it
+    if (isset($attributes['centre_id'])) {
+        $centre = App\Centre::find($attributes['centre_id']);
```

Comment:
Do we need to create one if not found or fallback to null?

Reply:
centre_id cannot be null; it's the centre the bundle was made at.

Reply:
I meant if the lookup fails - for bad data reasons... but I guess that will fall through to the next part with a `(!$centre)`